### PR TITLE
Add `pgroonga_crash_safer.max_recovery_threads` parameter

### PR DESCRIPTION
**Pull request by abetomo**
https://github.com/pgroonga/pgroonga/issues/397#issuecomment-1952009012
I added the parameters mentioned in that comment.

### CODE REVIEW

**Comment by abetomo** [comment]
The test failed and I will check it.

**Comment by abetomo** [comment]
The test that failed was Groonga 13.1.1.
`grn_set_default_n_workers` has not been added yet.

File: src/pgroonga-crash-safer.c [hunk 1]
```diff
@@ -845,6 +851,21 @@ _PG_init(void)
 							 NULL,
 							 NULL);
 
+	DefineCustomIntVariable("pgroonga_crash_safer.max_recovery_threads",
+							"Maximum number of threads for recovery of broken Groonga indexes.",
+							"The default is 0, which means disabled. "
+							"Use all CPUs in the environment at -1. "
+							"Use CPU for that number if max_recovery_threads >= 1 is set.",
```
Comment (kou) [inline src/pgroonga-crash-safer.c:858]:
```suggestion
							"Use CPU for that number if 1 or later is set.",
```

or `pgroogna_crash_safer.max_recovery_threads`.

File: test/test-pgroonga-crash-safer.rb [hunk 2]
```diff
@@ -282,4 +282,26 @@ def additional_reference_configurations
       end
     end
   end
+
+  test "max_recovery_threads parameter: default" do
```
Comment (kou) [inline test/test-pgroonga-crash-safer.rb:286]:
```suggestion
  test "max_recovery_threads: default" do
```

File: test/test-pgroonga-crash-safer.rb [hunk 3]
```diff
@@ -282,4 +282,26 @@ def additional_reference_configurations
...
+    pgroonga_log = @postgresql.read_pgroonga_log
+    assert_equal(["pgroonga: crash-safer: max_recovery_threads: 0"],
+                 pgroonga_log.scan(/pgroonga: crash-safer: max_recovery_threads:.*$/),
+                 pgroonga_log)
+  end
+
+  sub_test_case("max_recovery_threads parameter") do
```
Comment (kou) [inline test/test-pgroonga-crash-safer.rb:293]:
```suggestion
  sub_test_case("max_recovery_threads") do
```

File: test/test-pgroonga-crash-safer.rb [hunk 4]
```diff
@@ -282,4 +282,26 @@ def additional_reference_configurations
...
+    def additional_configurations
+      super + <<-CONFIG
+pgroonga_crash_safer.max_recovery_threads = -1
+      CONFIG
+    end
+
+    test "set -1" do
```
Comment (kou) [inline test/test-pgroonga-crash-safer.rb:300]:
```suggestion
    test "set" do
```

**Review by kou — COMMENTED** [review]
We need to update required Groonga version:

```diff
diff --git a/makefiles/pgrn-pgxs.mk b/makefiles/pgrn-pgxs.mk
index e106ed1e..7c28445b 100644
--- a/makefiles/pgrn-pgxs.mk
+++ b/makefiles/pgrn-pgxs.mk
@@ -1,4 +1,4 @@
-REQUIRED_GROONGA_VERSION = 13.0.2
+REQUIRED_GROONGA_VERSION = 14.0.0
 GROONGA_PKG = "groonga >= $(REQUIRED_GROONGA_VERSION)"
 PACKAGES += $(GROONGA_PKG)
 
```

**Pull request merged**